lib/init/grass.py

```diff
@@ -1900,10 +1900,21 @@ def csh_startup(location, location_name, mapset, grass_env_file):
     os.environ['HOME'] = userhome
     return process
 
+def shell_startup(location, location_name, grass_env_file, shell_name):
```

Comment:
In light of #768, `shell` should probably mean any shell in grass.py. Could this be `bourne_like_startup` or `sh_like_startup` or `sh_startup`?

Reply:
I agree with @wenzeslaus. `shell_startup` is too inclusive when this function only handles `bash` and `zsh`. There is `csh_startup` for another shell already, so again, semantically, `shell_startup` is not good either. If you felt awkward passing a `zsh` flag to `bash_startup` (which I'm OK with), maybe, simply `sh_startup` would be better.

However, `/bin/sh` (true `sh` shell) actually uses `default_startup`, `csh` and `tcsh` share `csh_startup`, `bash` and `msh` share `bash_startup`. Admittedly, these shells don't pass a shell-specific flag like `zsh`.

What about `bash_like_startup` or `bash_zsh_startup` (`bazsh_startup`.. let's be more creative)? It may be semantically more correct. There is no perfect solution when you combine code for multiple shells.

Reply:
> There is no perfect solution when you combine code for multiple shells.

Soo, true! Considering all the alternatives you mentioned and what I in addition could think of, may I suggest we choose `sh_startup`?
(Although bazsh_startup was not bad either, albeit a bit cryptic 😄)

Reply:
> Soo, true! Considering all the alternatives you mentioned and what I in addition could think of, may I suggest we choose `sh_startup`?
> (Although bazsh_startup was not bad either, albeit a bit cryptic smile)

`bazsh_startup` was a joke ;-). My only problem with `sh_startup` is that `sh` doesn't use `sh_startup`. I believe, like @wenzeslaus mentioned, `sh_like_startup` would be better in this case.

Reply:
I would rename `shell_name` to `sh` to be consistent with the rest of the script. `shellname` is used for the full name of a shell outside and `sh` for a short name. `shell_name` is the same as `sh`.

Reply:
> My only problem with sh_startup is that sh doesn't use sh_startup. I believe, like @wenzeslaus mentioned, sh_like_startup would be better in this case.

I can live with `sh_like_startup`.

Sidenote: I did notice that Bourne shell (sh) is using `default startup` and does not have a "fake" home in $LOCATION (hence no separate history for each location). Is this intended design or has there been no demand for this?

Reply:
I guess it's the latter. No demand.